crates/ruff_linter/src/rules/airflow/rules/removal_in_3.rs

```diff
@@ -849,3 +954,55 @@ fn is_airflow_builtin_or_provider(segments: &[&str], module: &str, symbol_suffix
         _ => false,
     }
 }
+
+fn is_task_context_referenced(checker: &mut Checker, expr: &Expr) -> bool {
```

Comment:
Can you clarify what this function is suppose to do? I'm confused as to why this function is also looping over the `REMOVED_CONTEXT_ARGS` and adding the diagnostics when the callee also does something similar. Additionally, the name and the return type of this function suggests that it checks for a condition but it's adding diagnostics as well.

Reply:
`is_task_context_referenced` had multiple responsibilities: check task context usage, add diagnostics and return a boolean. The name of the function was misleading.

I have now split the function to check whether any task context arguments reference removed keys and a separate method to add diagnostics. I named first method to `uses_removed_context_keys`.